### PR TITLE
Problem: eth signer not accurate in state machine

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- cleanup the eth signer related codes
- state machine always use `ethtypes.MakeSigner` based on chain config and block number, it matters when we do block number based hardfork and the signer logic is changed in that, it don't matter in `x/upgrade` style upgrade though.
- optimize rpc api to reuse the `From` field in msg, instead of always recover the sender.

Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -65,6 +65,8 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (evm) [#454](https://github.com/crypto-org-chain/ethermint/pull/454) Migrate transient stores to object stores.
 * (evm) [#461](https://github.com/crypto-org-chain/ethermint/pull/461) Remove custom signer getter, use the option in protobuf file instead.
 * (evm) [#469](https://github.com/crypto-org-chain/ethermint/pull/469) More aggressive cache of common parameters in object store.
+* (evm) [#]() State machine always use `ethtypes.MakeSigner` based on current chain config and block number, this is
```
Comment (yihuang) [inline CHANGELOG.md]:
```suggestion
* (evm) [#470](https://github.com/crypto-org-chain/ethermint/pull/470) State machine always use `ethtypes.MakeSigner` based on current chain config and block number, this is
```

File: rpc/backend/utils.go [hunk 2]
```diff
@@ -104,7 +104,7 @@ func (b *Backend) getAccountNonce(accAddr common.Address, pending bool, height i
 				break
 			}
 
-			sender, err := ethMsg.GetSenderLegacy(b.chainID)
+			sender, err := ethMsg.GetSenderLegacy(ethtypes.LatestSignerForChainID(b.chainID))
```
Comment (mmsqe) [inline rpc/backend/utils.go:107]: should we use MakeSigner as well

File: rpc/types/utils.go [hunk 3]
```diff
@@ -167,27 +167,30 @@ func NewTransactionFromMsg(
...
-	// transactions. For non-protected transactions, the homestead signer signer is used
-	// because the return value of ChainId is zero for those transactions.
+	// transactions. For non-protected transactions, the frontier signer is used
+	// because the latest signer will reject the unprotected transactions.
 	var signer ethtypes.Signer
 	if tx.Protected() {
 		signer = ethtypes.LatestSignerForChainID(tx.ChainId())
 	} else {
-		signer = ethtypes.HomesteadSigner{}
+		signer = ethtypes.FrontierSigner{}
```
Comment (mmsqe) [inline rpc/types/utils.go:188]: oh does this check wrong from beginning?

Comment (yihuang) [reply rpc/types/utils.go:188]: i think so

File: rpc/backend/utils.go (hunk 2 above)
Comment (yihuang) [reply rpc/backend/utils.go:107]: it's fine in json-rpc I think.

File: rpc/types/utils.go (hunk 3 above)
Comment (yihuang) [reply rpc/types/utils.go:188]: we are still not consistency in handling the legacy transactions across the json-rpc, anyway 🤦

**Review by mmsqe — APPROVED** [review]

**Pull request merged**